src/pybind/mgr/orchestrator.py

```diff
@@ -716,6 +718,69 @@ def __init__(self):
         # some replicaset special sauce for autoscaling?
         self.extended = {}
 
+        # Object with specific settings for the service
+        self.service_spec = None
+
+class RGWSpec(object):
+    """
+    Settings to configure a multisite Ceph RGW
+    """
+    def __init__(self, hosts=None, rgw_multisite=True, rgw_zone="Default_Zone",
+              rgw_zonemaster=True, rgw_zonesecondary=False,
+              rgw_multisite_proto="http", rgw_frontend_port="8080",
+              rgw_zonegroup="Main", rgw_zone_user="zone.user",
+              rgw_realm="RGW_Realm", system_access_key=None,
+              system_secret_key=None):
+
+        self.hosts = hosts
+        self.rgw_multisite=rgw_multisite
+        self.rgw_zone = rgw_zone
+        self.rgw_zonemaster = rgw_zonemaster
+        self.rgw_zonesecondary = rgw_zonesecondary
+        self.rgw_multisite_proto = rgw_multisite_proto
+        self.rgw_frontend_port = rgw_frontend_port
+
+        if hosts:
+            self.rgw_multisite_endpoint_addr = hosts[0]
+
+            self.rgw_multisite_endpoints_list = ",".join(
+                ["{}://{}:{}".format(self.rgw_multisite_proto,
+                                    host,
+                                    self.rgw_frontend_port) for host in hosts])
+
+        self.rgw_zonegroup = rgw_zonegroup
+        self.rgw_zone_user = rgw_zone_user
+        self.rgw_realm = rgw_realm
+
+        if system_access_key:
+            self.system_access_key = system_access_key
+        else:
+            self.system_access_key = self.genkey(20)
+        if system_secret_key:
+            self.system_secret_key = system_secret_key
+        else:
+            self.system_secret_key = self.genkey(40)
+
+    def genkey(self, nchars):
+        """ Returns a random string of nchars
+
+        :nchars : Length of the returned string
+        """
+
+        return ''.join(random.choice(string.ascii_uppercase +
```

Comment:
Just for your info: I just checked and it seems that RGW only creates keys with uppercase + numbers (you may look for `gen_rand_alphanumeric_upper` method in `rgw_common.cc`). I don't it's relevant, but just for you to know.

Reply:
ouch... i will extrapolate this from the RH doc for RGW service... but better to check code!
Thanks. I will remove lower case letters in the generation.

Reply:
No, no, I don't mean this is wrong or that you need to change it. Probably it works perfectly, just wanted to note that it's different in RGW code.

Reply:
can we postpone this? As I said I'd like to refactor this in a different PR